3-networks/modules/vpn-ha/main.tf

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/******************************************
+  HA VPN configuration
+ *****************************************/
+
+locals {
+  vpc_name     = "${var.environment_code}-shared-${var.vpc_label}"
+  network_name = "vpc-${local.vpc_name}"
+}
+
+
+module "vpn_ha_region1_router1" {
+  source     = "terraform-google-modules/vpn/google//modules/vpn_ha"
+  project_id = var.project_id
+  region     = var.default_region1
+  network    = local.network_name
+  name       = "vpn-${local.vpc_name}-${var.default_region1}-cr1"
+  peer_external_gateway = {
+    redundancy_type = "TWO_IPS_REDUNDANCY"
+    interfaces = [{
+      id         = 0
+      ip_address = var.on_prem_router_ip_address1
+      },
+      {
+        id         = 1
+        ip_address = var.on_prem_router_ip_address2
+    }]
+  }
+  router_name = "cr-${local.vpc_name}-${var.default_region1}-cr1"
```

Comment:
The `router_name` could be received as an explicit variable since it should exist and have been already created in the same context this module will be called.

Reply:
Yes :) Already fixed